### PR TITLE
feat: add scrolling fps e2e test

### PR DESCRIPTION
**Pull request by lumixraku**
close https://github.com/dream-num/univer-pro/issues/2600

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: e2e/perf/scroll.spec.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    avgFps: number;
+}
+
+interface IJsonObject {
+    [key: string]: any;
+}
+
+interface IMeasureFPSParam { duration: number; deltaX: number; deltaY: number }
+test.beforeEach(async ({ context }) => {
+    await context.addInitScript(() => {
```
Comment (wzhudev) [inline e2e/perf/scroll.spec.ts:34]: Remove empty logic.

File: e2e/perf/scroll.spec.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                    cancelable: true,
+                    deltaY,
+                    deltaX,
+                    clientX: 580,
+                    clientY: 580,
+                });
+                element.dispatchEvent(event);
+            };
+
+            // 持续模拟 wheel 事件
```
Comment (wzhudev) [inline e2e/perf/scroll.spec.ts:68]: Use English in comments.

File: e2e/perf/scroll.spec.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    return imgData[3] !== 0;
+                }
+            });
+        });
+
+        try {
+            const fps = await measureFPS(page, 10, 0, 100);
+            await test.step('fps', async () => {
+                console.log('fps', fps);
+                expect(fps).toBeGreaterThan(minFpsValue);
```
Comment (wzhudev) [inline e2e/perf/scroll.spec.ts:132]: Not only should we monitor average FPS, but also also p99.

File: e2e/perf/scroll.spec.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            });
+        } catch (error) {
+            console.error('error when exec scrolling test', error);
+        } finally {
+            // Keep page
```
Comment (wzhudev) [inline e2e/perf/scroll.spec.ts:137]: Remove dead code.

File: e2e/perf/scroll.spec.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                    const imgData = canvaslist[2]!.getContext('2d')!.getImageData(40, 40, 1, 1).data;
+                    return imgData[3] !== 0;
+                }
+            });
+        });
+
+        try {
+            const fps = await measureFPS(page, 10, 0, 100);
+            await test.step('fps', async () => {
+                console.log('fps', fps);
```
Comment (wzhudev) [inline e2e/perf/scroll.spec.ts:131]: We should track this in a time sequence. Maybe store it to another system? cc @jikkai

Comment (lumixraku) [reply e2e/perf/scroll.spec.ts:134]: @jikkai  

The current output format is as shown below; how should I output it to make extraction easier in future?


image

Comment (lumixraku) [reply e2e/perf/scroll.spec.ts]: image

Comment (lumixraku) [reply e2e/perf/scroll.spec.ts]: image

Comment (lumixraku) [reply e2e/perf/scroll.spec.ts]: Add max 10 frame time

**Review by wzhudev — APPROVED** [review]

**Pull request merged**